contracts/EthStorageContract.sol

```diff
@@ -139,25 +138,17 @@ contract EthStorageContract is StorageContract, Decoder {
     /*
      * Decode the sample and check the decoded sample is included in the BLOB corresponding to on-chain datahashes.
      */
-    function decodeAndCheckInclusive(
+    function checkInclusive(
         uint256 kvIdx,
         uint256 sampleIdxInKv,
-        address miner,
         bytes32 encodedData,
         bytes calldata proof
-    ) public view virtual override returns (bool) {
+    ) public view virtual override returns (bool, uint256) {
         PhyAddr memory kvInfo = kvMap[idxMap[kvIdx]];
-        (Proof memory decodeProof, uint256 mask, bytes memory peInput) = abi.decode(proof, (Proof, uint256, bytes));
-
-        // BLOB decoding check
-        if (
-            !decodeSample(decodeProof, uint256(keccak256(abi.encode(kvInfo.hash, miner, kvIdx))), sampleIdxInKv, mask)
-        ) {
-            return false;
-        }
+        (uint256 mask, bytes memory peInput) = abi.decode(proof, (uint256, bytes));
```

Comment:
Could it be better to decode earlier, i.e., moving decode to the loop of  `_verifySamples`, for returning mask is not like a result `checkInclusive` should have.